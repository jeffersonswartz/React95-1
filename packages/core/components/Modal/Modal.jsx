import React, { useState, useContext, useEffect, forwardRef } from 'react';
import PropTypes from 'prop-types';
import styled, { css } from '@xstyled/styled-components';
import { th, backgroundColor } from '@xstyled/system';
import Draggable from 'react-draggable';

import Btn from '../shared-style/Btn';
import Button from '../Button';
import Icon from '../Icon';
import ModalContext from './ModalContext';

const ModalWrapper = styled.div`
  display: flex;
  flex-direction: column;

  position: fixed;

  padding: 2 2 8;

  top: 50px;

  background-color: bg;

  box-shadow: inset 1px 1px 0px 1px ${th('colors.white')},
    inset 0 0 0 1px ${th('colors.grays.3')}, 1px 1px 0 1px ${th('colors.black')};

  ${({ width, height }) => `
    width: ${width ? `${width}px` : 'auto'};
    height: ${height ? `${height}px` : 'auto'};
  `}
  ${({ active }) =>
    active
      ? css`
          z-index: modal;
        `
      : ''}
`;

const TitleBar = styled.div`
  height: 18px;
  margin-bottom: 2;

  color: ${th('colors.white')};
  padding: 2 2 0;

  display: flex;
  ${backgroundColor}
`;

const Title = styled.div`
  flex-grow: 1;
  font-weight: bold;
`;

const OptionsBox = styled.ul`
  list-style: none;
  padding: 0;
  margin: 0;

  display: flex;
`;

const Option = styled(Btn)`
  margin-right: 2;
  padding: 0;

  width: 17px;
  height: 14px;
  min-width: 0;

  font-size: 10;

  &:last-child {
    margin-right: 0;
  }

  &:active {
    padding: 1 0 0 1;

    outline: none;
  }

  &:focus {
    box-shadow: inset 1px 1px 0px 1px ${th('colors.white')},
      inset -1px -1px 0px 1px ${th('colors.grays.3')};
  }
`;

Option.displayName = 'Option';

const Content = styled.div`
  flex-grow: 1;
  display: flex;
  flex-direction: column;

  padding: 6;
`;

const ButtonWrapper = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: ${props => props.buttonsAlignment};

  padding: 0 6 6 6;

  & ${Btn} {
    margin-right: 6;
    min-width: 70px;

    &:last-child {
      margin-right: 0;
    }
  }
`;

const MenuWrapper = styled.ul`
  display: flex;
  flex-direction: row;

  list-style: none;
  margin: 0;
  padding-left: 0;
  padding-bottom: 3;

  border-bottom-style: solid;
  border-width: 1;
  border-bottom-color: grays.3;

  box-shadow: 0 1px 0 0 ${th('colors.grays.0')};
`;

const MenuItem = styled.li`
  position: relative;
  padding-left: 6;
  padding-right: 6;

  user-select: none;

  ul {
    position: absolute;
    left: 0;
    color: ${th('colors.black')};
  }

  ${({ active }) =>
    active &&
    css`
      background-color: primary;
      color: ${th('colors.white')};
    `};
`;

MenuItem.displayName = 'MenuItem';

const Modal = forwardRef(
  (
    {
      closeModal,
      title,
      children,
      buttons,
      icon,
      menu,
      buttonsAlignment,
      defaultPosition,
      width,
      height,
      ...rest
    },
    ref,
  ) => {
    const {
      addWindows,
      removeWindows,
      setActiveWindow,
      activeWindow,
    } = useContext(ModalContext);
    const [menuOpened, setMenuOpened] = useState('');

    useEffect(() => {
      addWindows({ icon, title });
      setActiveWindow(title);

      return () => removeWindows(title);
    }, []);

    const iconStyle = {
      width: 15,
      height: 13,
      style: {
        marginRight: '4px',
      },
    };

    const isActive = title === activeWindow;

    return (
      <>
        <Draggable handle=".draggable" defaultPosition={defaultPosition}>
          <ModalWrapper
            width={width}
            height={height}
            {...rest}
            onClick={() => setActiveWindow(title)}
            active={isActive}
            ref={ref}
          >
            <TitleBar
              className="draggable"
              backgroundColor={isActive ? 'primary' : 'grays.3'}
            >
              {icon && <Icon name={icon} {...iconStyle} />}
              <Title>{title}</Title>
              <OptionsBox>
                <Option>?</Option>
                <Option onClick={closeModal}>x</Option>
              </OptionsBox>
            </TitleBar>

            {menu.length > 0 && (
              <MenuWrapper>
                {menu.map(({ name, list }) => {
                  const active = menuOpened === name;
                  return (
                    <MenuItem
                      key={name}
                      onMouseDown={() => setMenuOpened(name)}
                      active={active}
                    >
                      {name}
                      {active && list}
                    </MenuItem>
                  );
                })}
              </MenuWrapper>
            )}

            <Content onClick={() => setMenuOpened('')}>{children}</Content>
            {buttons.length > 0 && (
              <ButtonWrapper buttonsAlignment={buttonsAlignment}>
                {buttons.map(button => (
                  <Button
                    key={button.value}
                    onClick={button.onClick}
                    value={button.value}
                  />
                ))}
              </ButtonWrapper>
            )}
          </ModalWrapper>
        </Draggable>
      </>
    );
  },
);

Modal.displayName = 'Modal';

Modal.propTypes = {
  icon: PropTypes.string,
  closeModal: PropTypes.func,
  title: PropTypes.string,
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]),
  buttons: PropTypes.arrayOf(
    PropTypes.shape({
      value: PropTypes.string.isRequired,
      onClick: PropTypes.func,
    }),
  ),
  menu: PropTypes.arrayOf(
    PropTypes.shape({
      name: PropTypes.string.isRequired,
      list: PropTypes.oneOfType([
        PropTypes.arrayOf(PropTypes.node),
        PropTypes.node,
      ]).isRequired,
    }),
  ),
  buttonsAlignment: PropTypes.oneOf(['center', 'flex-start', 'flex-end']),
  defaultPosition: PropTypes.shape({
    x: PropTypes.number.isRequired,
    y: PropTypes.number.isRequired,
  }),
  width: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
  height: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
};

Modal.defaultProps = {
  icon: '',
  title: 'Modal',
  buttonsAlignment: 'flex-end',
  children: null,
  defaultPosition: { x: 0, y: 0 },
  buttons: [],
  menu: [],
  width: undefined,
  height: undefined,
  closeModal: () => {},
};

export default Modal;
