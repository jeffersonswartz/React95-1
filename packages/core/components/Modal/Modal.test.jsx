import React from 'react';
import { waitRender, fireEvent } from '../shared/test/utils';
import Modal from './Modal';
import ModalContext from './ModalContext';
import List from '../List';

describe('<Modal />', () => {
  describe('Snapshots', () => {
    it('should match snapshot', async () => {
      const { container } = await waitRender(
        <Modal
          icon="bat"
          title="file.bat"
          buttons={[{ value: 'Ok' }, { value: 'Cancel' }]}
          menu={[
            {
              name: 'File',
              list: (
                <List>
                  <List.Item>Exit</List.Item>
                </List>
              ),
            },
            {
              name: 'Edit',
              list: (
                <List>
                  <List.Item>Copy</List.Item>
                </List>
              ),
            },
          ]}
        >
          Hello
        </Modal>,
      );
      expect(container).toMatchSnapshot();
    });

    it('should match snapshot with buttonsAligment prop equals center', async () => {
      const { container } = await waitRender(
        <Modal
          icon="bat"
          title="file.bat"
          buttons={[{ value: 'Ok' }, { value: 'Cancel' }]}
          buttonsAlignment="center"
        >
          Hello
        </Modal>,
      );
      expect(container).toMatchSnapshot();
    });

    it('should match snapshot with buttonsAligment prop equals flex-start', async () => {
      const { container } = await waitRender(
        <Modal
          icon="bat"
          title="file.bat"
          buttons={[{ value: 'Ok' }, { value: 'Cancel' }]}
          buttonsAlignment="flex-start"
        >
          Hello
        </Modal>,
      );
      expect(container).toMatchSnapshot();
    });

    it('should match snapshot with width and height props', async () => {
      const { container } = await waitRender(
        <Modal icon="bat" title="file.bat" width={300} height={200}>
          Hello
        </Modal>,
      );
      expect(container).toMatchSnapshot();
    });

    it('should match snapshot with different activeWindow', async () => {
      const { container } = await waitRender(
        <ModalContext.Provider
          value={{
            windows: [],
            addWindows: () => {},
            removeWindows: () => {},
            setActiveWindow: () => {},
            activeWindow: '',
          }}
        >
          <Modal icon="bat" title="file.bat" width={300} height={200}>
            Hello
          </Modal>
        </ModalContext.Provider>,
      );
      expect(container).toMatchSnapshot();
    });
  });

  describe('closeModal prop', () => {
    it('should call closeModal when Modal close button is clicked', async () => {
      const closeModalMock = jest.fn();
      const { getByText } = await waitRender(
        <Modal icon="bat" title="file.bat" closeModal={closeModalMock}>
          Hello
        </Modal>,
      );

      fireEvent.click(getByText('x'));

      expect(closeModalMock).toHaveBeenCalled();
    });
  });

  describe('Modal action buttons', () => {
    it('should display button text correctly', async () => {
      const buttonText = 'button text';
      const { container } = await waitRender(
        <Modal icon="bat" title="file.bat" buttons={[{ value: buttonText }]}>
          Hello
        </Modal>,
      );

      expect(
        container.querySelector(`button[value='${buttonText}']`).textContent,
      ).toBe(buttonText);
    });

    it('should display more than one button', async () => {
      const { getAllByText } = await waitRender(
        <Modal
          icon="bat"
          title="file.bat"
          buttons={[
            { value: 'button' },
            { value: 'button' },
            { value: 'button' },
          ]}
        >
          Hello
        </Modal>,
      );

      expect(getAllByText('button').length).toBe(3);
    });

    it('should call onClick function when Modal action button is clicked', async () => {
      const onClickMock = jest.fn();
      const { getByText } = await waitRender(
        <Modal buttons={[{ value: 'Ok', onClick: onClickMock }]}>Hello</Modal>,
      );

      fireEvent.click(getByText('Ok'));

      expect(onClickMock).toHaveBeenCalled();
    });
  });

  describe('Menus', () => {
    it('should display menu name correctly', async () => {
      const { container } = await waitRender(
        <Modal
          menu={[
            {
              name: 'Menu Text',
              list: (
                <List>
                  <List.Item>Exit</List.Item>
                </List>
              ),
            },
          ]}
        >
          Hello
        </Modal>,
      );

      expect(container.querySelector('li').textContent).toBe('Menu Text');
    });

    it('should display menu list correctly', async () => {
      const { getByText } = await waitRender(
        <Modal
          menu={[
            {
              name: 'Edit',
              list: (
                <List>
                  <List.Item>Exit</List.Item>
                </List>
              ),
            },
          ]}
        >
          Hello
        </Modal>,
      );

      fireEvent.mouseDown(getByText('Edit'));

      expect(getByText('Edit').querySelector('li').textContent).toBe('Exit');
    });
  });
});
