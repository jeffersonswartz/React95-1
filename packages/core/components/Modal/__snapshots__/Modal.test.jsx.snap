// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Modal /> Snapshots should match snapshot 1`] = `
.c11 {
  background-color: #c3c7cb;
  padding: 7px 20px 5px;
  border: none;
  font-size: 12px;
  box-shadow: inset 1px 1px 0px 1px #ffffff, inset 0 0 0 1px #868a8e,1px 1px 0 0px #000000;
}

.c11:disabled {
  color: #868a8e;
}

.c11:focus {
  outline: 1px dotted #000000;
  outline-offset: -5px;
  box-shadow: inset 1px 1px 0px 1px #ffffff, inset -0.5px -0.5px 0px 1px #868a8e, 1px 1px 0 1px #000000;
}

.c11:active {
  padding: 8px 20px 4px;
  outline: 1px dotted #000000;
  outline-offset: -5px;
  box-shadow: inset 0 0 0 1px #868a8e, 0 0 0 1px #000000;
}

.c2 {
  display: block;
  background-repeat: no-repeat;
  background-position: center;
  background-size: contain;
  width: 15px;
  height: 13px;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  position: fixed;
  padding: 2px 2px 8px;
  top: 50px;
  background-color: #c3c7cb;
  box-shadow: inset 1px 1px 0px 1px #ffffff, inset 0 0 0 1px #868a8e,1px 1px 0 1px #000000;
  width: auto;
  height: auto;
  z-index: 1;
}

.c1 {
  height: 18px;
  margin-bottom: 2px;
  color: #ffffff;
  padding: 2px 2px 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  background-color: #000e7a;
}

.c3 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  font-weight: bold;
}

.c4 {
  list-style: none;
  padding: 0;
  margin: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c6 {
  background-color: #c3c7cb;
  padding: 7px 20px 5px;
  border: none;
  font-size: 12px;
  box-shadow: inset 1px 1px 0px 1px #ffffff, inset 0 0 0 1px #868a8e,1px 1px 0 0px #000000;
  margin-right: 2px;
  padding: 0;
  width: 17px;
  height: 14px;
  min-width: 0;
  font-size: 10px;
}

.c6:disabled {
  color: #868a8e;
}

.c6:focus {
  outline: 1px dotted #000000;
  outline-offset: -5px;
  box-shadow: inset 1px 1px 0px 1px #ffffff, inset -0.5px -0.5px 0px 1px #868a8e, 1px 1px 0 1px #000000;
}

.c6:active {
  padding: 8px 20px 4px;
  outline: 1px dotted #000000;
  outline-offset: -5px;
  box-shadow: inset 0 0 0 1px #868a8e, 0 0 0 1px #000000;
}

.c6:last-child {
  margin-right: 0;
}

.c6:active {
  padding: 1px 0 0 1px;
  outline: none;
}

.c6:focus {
  box-shadow: inset 1px 1px 0px 1px #ffffff, inset -1px -1px 0px 1px #868a8e;
}

.c9 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 6px;
}

.c10 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  padding: 0 6px 6px 6px;
}

.c10 .c5 {
  margin-right: 6px;
  min-width: 70px;
}

.c10 .c5:last-child {
  margin-right: 0;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  list-style: none;
  margin: 0;
  padding-left: 0;
  padding-bottom: 3px;
  border-bottom-style: solid;
  border-width: 1px;
  border-bottom-color: #868a8e;
  box-shadow: 0 1px 0 0 #e6e6e6;
}

.c8 {
  position: relative;
  padding-left: 6px;
  padding-right: 6px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c8 ul {
  position: absolute;
  left: 0;
  color: #000000;
}

<div>
  <div
    class="c0 react-draggable"
    style="transform: translate(0px,0px);"
  >
    <div
      class="c1 draggable"
    >
      <i
        class="c2"
        height="13"
        name="bat"
        style="margin-right: 4px; background-image: url(buffer-32-4-1);"
        width="15"
      />
      <div
        class="c3"
      >
        file.bat
      </div>
      <ul
        class="c4"
      >
        <button
          class="c5 c6"
        >
          ?
        </button>
        <button
          class="c5 c6"
        >
          x
        </button>
      </ul>
    </div>
    <ul
      class="c7"
    >
      <li
        class="c8"
      >
        File
      </li>
      <li
        class="c8"
      >
        Edit
      </li>
    </ul>
    <div
      class="c9"
    >
      Hello
    </div>
    <div
      class="c10"
    >
      <button
        class="c5 c11"
        value="Ok"
      >
        Ok
      </button>
      <button
        class="c5 c11"
        value="Cancel"
      >
        Cancel
      </button>
    </div>
  </div>
</div>
`;

exports[`<Modal /> Snapshots should match snapshot with buttonsAligment prop equals center 1`] = `
.c9 {
  background-color: #c3c7cb;
  padding: 7px 20px 5px;
  border: none;
  font-size: 12px;
  box-shadow: inset 1px 1px 0px 1px #ffffff, inset 0 0 0 1px #868a8e,1px 1px 0 0px #000000;
}

.c9:disabled {
  color: #868a8e;
}

.c9:focus {
  outline: 1px dotted #000000;
  outline-offset: -5px;
  box-shadow: inset 1px 1px 0px 1px #ffffff, inset -0.5px -0.5px 0px 1px #868a8e, 1px 1px 0 1px #000000;
}

.c9:active {
  padding: 8px 20px 4px;
  outline: 1px dotted #000000;
  outline-offset: -5px;
  box-shadow: inset 0 0 0 1px #868a8e, 0 0 0 1px #000000;
}

.c2 {
  display: block;
  background-repeat: no-repeat;
  background-position: center;
  background-size: contain;
  width: 15px;
  height: 13px;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  position: fixed;
  padding: 2px 2px 8px;
  top: 50px;
  background-color: #c3c7cb;
  box-shadow: inset 1px 1px 0px 1px #ffffff, inset 0 0 0 1px #868a8e,1px 1px 0 1px #000000;
  width: auto;
  height: auto;
  z-index: 1;
}

.c1 {
  height: 18px;
  margin-bottom: 2px;
  color: #ffffff;
  padding: 2px 2px 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  background-color: #000e7a;
}

.c3 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  font-weight: bold;
}

.c4 {
  list-style: none;
  padding: 0;
  margin: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c6 {
  background-color: #c3c7cb;
  padding: 7px 20px 5px;
  border: none;
  font-size: 12px;
  box-shadow: inset 1px 1px 0px 1px #ffffff, inset 0 0 0 1px #868a8e,1px 1px 0 0px #000000;
  margin-right: 2px;
  padding: 0;
  width: 17px;
  height: 14px;
  min-width: 0;
  font-size: 10px;
}

.c6:disabled {
  color: #868a8e;
}

.c6:focus {
  outline: 1px dotted #000000;
  outline-offset: -5px;
  box-shadow: inset 1px 1px 0px 1px #ffffff, inset -0.5px -0.5px 0px 1px #868a8e, 1px 1px 0 1px #000000;
}

.c6:active {
  padding: 8px 20px 4px;
  outline: 1px dotted #000000;
  outline-offset: -5px;
  box-shadow: inset 0 0 0 1px #868a8e, 0 0 0 1px #000000;
}

.c6:last-child {
  margin-right: 0;
}

.c6:active {
  padding: 1px 0 0 1px;
  outline: none;
}

.c6:focus {
  box-shadow: inset 1px 1px 0px 1px #ffffff, inset -1px -1px 0px 1px #868a8e;
}

.c7 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 6px;
}

.c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0 6px 6px 6px;
}

.c8 .c5 {
  margin-right: 6px;
  min-width: 70px;
}

.c8 .c5:last-child {
  margin-right: 0;
}

<div>
  <div
    class="c0 react-draggable"
    style="transform: translate(0px,0px);"
  >
    <div
      class="c1 draggable"
    >
      <i
        class="c2"
        height="13"
        name="bat"
        style="margin-right: 4px; background-image: url(buffer-32-4-1);"
        width="15"
      />
      <div
        class="c3"
      >
        file.bat
      </div>
      <ul
        class="c4"
      >
        <button
          class="c5 c6"
        >
          ?
        </button>
        <button
          class="c5 c6"
        >
          x
        </button>
      </ul>
    </div>
    <div
      class="c7"
    >
      Hello
    </div>
    <div
      class="c8"
    >
      <button
        class="c5 c9"
        value="Ok"
      >
        Ok
      </button>
      <button
        class="c5 c9"
        value="Cancel"
      >
        Cancel
      </button>
    </div>
  </div>
</div>
`;

exports[`<Modal /> Snapshots should match snapshot with buttonsAligment prop equals flex-start 1`] = `
.c9 {
  background-color: #c3c7cb;
  padding: 7px 20px 5px;
  border: none;
  font-size: 12px;
  box-shadow: inset 1px 1px 0px 1px #ffffff, inset 0 0 0 1px #868a8e,1px 1px 0 0px #000000;
}

.c9:disabled {
  color: #868a8e;
}

.c9:focus {
  outline: 1px dotted #000000;
  outline-offset: -5px;
  box-shadow: inset 1px 1px 0px 1px #ffffff, inset -0.5px -0.5px 0px 1px #868a8e, 1px 1px 0 1px #000000;
}

.c9:active {
  padding: 8px 20px 4px;
  outline: 1px dotted #000000;
  outline-offset: -5px;
  box-shadow: inset 0 0 0 1px #868a8e, 0 0 0 1px #000000;
}

.c2 {
  display: block;
  background-repeat: no-repeat;
  background-position: center;
  background-size: contain;
  width: 15px;
  height: 13px;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  position: fixed;
  padding: 2px 2px 8px;
  top: 50px;
  background-color: #c3c7cb;
  box-shadow: inset 1px 1px 0px 1px #ffffff, inset 0 0 0 1px #868a8e,1px 1px 0 1px #000000;
  width: auto;
  height: auto;
  z-index: 1;
}

.c1 {
  height: 18px;
  margin-bottom: 2px;
  color: #ffffff;
  padding: 2px 2px 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  background-color: #000e7a;
}

.c3 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  font-weight: bold;
}

.c4 {
  list-style: none;
  padding: 0;
  margin: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c6 {
  background-color: #c3c7cb;
  padding: 7px 20px 5px;
  border: none;
  font-size: 12px;
  box-shadow: inset 1px 1px 0px 1px #ffffff, inset 0 0 0 1px #868a8e,1px 1px 0 0px #000000;
  margin-right: 2px;
  padding: 0;
  width: 17px;
  height: 14px;
  min-width: 0;
  font-size: 10px;
}

.c6:disabled {
  color: #868a8e;
}

.c6:focus {
  outline: 1px dotted #000000;
  outline-offset: -5px;
  box-shadow: inset 1px 1px 0px 1px #ffffff, inset -0.5px -0.5px 0px 1px #868a8e, 1px 1px 0 1px #000000;
}

.c6:active {
  padding: 8px 20px 4px;
  outline: 1px dotted #000000;
  outline-offset: -5px;
  box-shadow: inset 0 0 0 1px #868a8e, 0 0 0 1px #000000;
}

.c6:last-child {
  margin-right: 0;
}

.c6:active {
  padding: 1px 0 0 1px;
  outline: none;
}

.c6:focus {
  box-shadow: inset 1px 1px 0px 1px #ffffff, inset -1px -1px 0px 1px #868a8e;
}

.c7 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 6px;
}

.c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  padding: 0 6px 6px 6px;
}

.c8 .c5 {
  margin-right: 6px;
  min-width: 70px;
}

.c8 .c5:last-child {
  margin-right: 0;
}

<div>
  <div
    class="c0 react-draggable"
    style="transform: translate(0px,0px);"
  >
    <div
      class="c1 draggable"
    >
      <i
        class="c2"
        height="13"
        name="bat"
        style="margin-right: 4px; background-image: url(buffer-32-4-1);"
        width="15"
      />
      <div
        class="c3"
      >
        file.bat
      </div>
      <ul
        class="c4"
      >
        <button
          class="c5 c6"
        >
          ?
        </button>
        <button
          class="c5 c6"
        >
          x
        </button>
      </ul>
    </div>
    <div
      class="c7"
    >
      Hello
    </div>
    <div
      class="c8"
    >
      <button
        class="c5 c9"
        value="Ok"
      >
        Ok
      </button>
      <button
        class="c5 c9"
        value="Cancel"
      >
        Cancel
      </button>
    </div>
  </div>
</div>
`;

exports[`<Modal /> Snapshots should match snapshot with different activeWindow 1`] = `
.c2 {
  display: block;
  background-repeat: no-repeat;
  background-position: center;
  background-size: contain;
  width: 15px;
  height: 13px;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  position: fixed;
  padding: 2px 2px 8px;
  top: 50px;
  background-color: #c3c7cb;
  box-shadow: inset 1px 1px 0px 1px #ffffff, inset 0 0 0 1px #868a8e,1px 1px 0 1px #000000;
  width: 300px;
  height: 200px;
}

.c1 {
  height: 18px;
  margin-bottom: 2px;
  color: #ffffff;
  padding: 2px 2px 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  background-color: #868a8e;
}

.c3 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  font-weight: bold;
}

.c4 {
  list-style: none;
  padding: 0;
  margin: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c5 {
  background-color: #c3c7cb;
  padding: 7px 20px 5px;
  border: none;
  font-size: 12px;
  box-shadow: inset 1px 1px 0px 1px #ffffff, inset 0 0 0 1px #868a8e,1px 1px 0 0px #000000;
  margin-right: 2px;
  padding: 0;
  width: 17px;
  height: 14px;
  min-width: 0;
  font-size: 10px;
}

.c5:disabled {
  color: #868a8e;
}

.c5:focus {
  outline: 1px dotted #000000;
  outline-offset: -5px;
  box-shadow: inset 1px 1px 0px 1px #ffffff, inset -0.5px -0.5px 0px 1px #868a8e, 1px 1px 0 1px #000000;
}

.c5:active {
  padding: 8px 20px 4px;
  outline: 1px dotted #000000;
  outline-offset: -5px;
  box-shadow: inset 0 0 0 1px #868a8e, 0 0 0 1px #000000;
}

.c5:last-child {
  margin-right: 0;
}

.c5:active {
  padding: 1px 0 0 1px;
  outline: none;
}

.c5:focus {
  box-shadow: inset 1px 1px 0px 1px #ffffff, inset -1px -1px 0px 1px #868a8e;
}

.c6 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 6px;
}

<div>
  <div
    class="c0 react-draggable"
    height="200"
    style="transform: translate(0px,0px);"
    width="300"
  >
    <div
      class="c1 draggable"
    >
      <i
        class="c2"
        height="13"
        name="bat"
        style="margin-right: 4px; background-image: url(buffer-32-4-1);"
        width="15"
      />
      <div
        class="c3"
      >
        file.bat
      </div>
      <ul
        class="c4"
      >
        <button
          class="sc-pscky c5"
        >
          ?
        </button>
        <button
          class="sc-pscky c5"
        >
          x
        </button>
      </ul>
    </div>
    <div
      class="c6"
    >
      Hello
    </div>
  </div>
</div>
`;

exports[`<Modal /> Snapshots should match snapshot with width and height props 1`] = `
.c2 {
  display: block;
  background-repeat: no-repeat;
  background-position: center;
  background-size: contain;
  width: 15px;
  height: 13px;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  position: fixed;
  padding: 2px 2px 8px;
  top: 50px;
  background-color: #c3c7cb;
  box-shadow: inset 1px 1px 0px 1px #ffffff, inset 0 0 0 1px #868a8e,1px 1px 0 1px #000000;
  width: 300px;
  height: 200px;
  z-index: 1;
}

.c1 {
  height: 18px;
  margin-bottom: 2px;
  color: #ffffff;
  padding: 2px 2px 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  background-color: #000e7a;
}

.c3 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  font-weight: bold;
}

.c4 {
  list-style: none;
  padding: 0;
  margin: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c5 {
  background-color: #c3c7cb;
  padding: 7px 20px 5px;
  border: none;
  font-size: 12px;
  box-shadow: inset 1px 1px 0px 1px #ffffff, inset 0 0 0 1px #868a8e,1px 1px 0 0px #000000;
  margin-right: 2px;
  padding: 0;
  width: 17px;
  height: 14px;
  min-width: 0;
  font-size: 10px;
}

.c5:disabled {
  color: #868a8e;
}

.c5:focus {
  outline: 1px dotted #000000;
  outline-offset: -5px;
  box-shadow: inset 1px 1px 0px 1px #ffffff, inset -0.5px -0.5px 0px 1px #868a8e, 1px 1px 0 1px #000000;
}

.c5:active {
  padding: 8px 20px 4px;
  outline: 1px dotted #000000;
  outline-offset: -5px;
  box-shadow: inset 0 0 0 1px #868a8e, 0 0 0 1px #000000;
}

.c5:last-child {
  margin-right: 0;
}

.c5:active {
  padding: 1px 0 0 1px;
  outline: none;
}

.c5:focus {
  box-shadow: inset 1px 1px 0px 1px #ffffff, inset -1px -1px 0px 1px #868a8e;
}

.c6 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 6px;
}

<div>
  <div
    class="c0 react-draggable"
    height="200"
    style="transform: translate(0px,0px);"
    width="300"
  >
    <div
      class="c1 draggable"
    >
      <i
        class="c2"
        height="13"
        name="bat"
        style="margin-right: 4px; background-image: url(buffer-32-4-1);"
        width="15"
      />
      <div
        class="c3"
      >
        file.bat
      </div>
      <ul
        class="c4"
      >
        <button
          class="sc-pscky c5"
        >
          ?
        </button>
        <button
          class="sc-pscky c5"
        >
          x
        </button>
      </ul>
    </div>
    <div
      class="c6"
    >
      Hello
    </div>
  </div>
</div>
`;
